### PR TITLE
Add check for table reappear in tablegroups

### PR DESCRIPTION
**Pull request by Huy-DNA**
## Summary
* Table reappearing in different TableGroups will raise an error

## Issue
(issue link here)

## Lasting Changes (Technical)

* `TableGroupInterpreter`
* Interpreter's `env`

## Checklist

Please check directly on the box once each of these are done

- [X] Documentation (if necessary)
- [X] Tests (integration test/unit test)
- [ ] Integration Tests Passed
- [ ] Code Review

### CODE REVIEW

File: packages/dbml-parse/src/lib/interpreter/types.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ export interface InterpreterDatabase {
   refIds: { [refid: string]: ElementDeclarationNode };
   ref: Map<ElementDeclarationNode, Ref>;
   enums: Map<ElementDeclarationNode, Enum>;
+  // for keeping track of the owner table group of a table
+  tableOwnerGroup: { [tableid: string]: ElementDeclarationNode };
```
Comment (nguyenalter) [inline packages/dbml-parse/src/lib/interpreter/types.ts:22]: The current name is fine but I suggest the following names:
- groupOfTable
- groupNodeOfTable

**Review by nguyenalter — APPROVED** [review]
LGTM!

File: packages/dbml-parse/src/lib/interpreter/elementInterpreter/tableGroup.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import { destructureComplexVariable } from "../../analyzer/utils";
+import _ from "lodash";
```
[the remark this answers is not in the document]
Comment (Huy-DNA) [reply packages/dbml-parse/src/lib/interpreter/elementInterpreter/tableGroup.ts]: Fixed!

File: packages/dbml-parse/src/lib/interpreter/types.ts (hunk 1 above)
Comment (Huy-DNA) [reply packages/dbml-parse/src/lib/interpreter/types.ts]: Renamed to `groupOfTable`!

**Pull request merged**